src/Codegen/Types.hack

```diff
@@ -176,23 +175,26 @@ enum IO: string as string {
  * Get the type aias for the given type structure.
  */
 function type_structure_to_type_alias<T>(TypeStructure<T> $ts): string {
+    $prefix = ($ts['nullable'] ?? false) ? '?' : '';
+
     $alias = Shapes::idx($ts, 'alias');
     if ($alias is nonnull) {
-        return $alias;
+        return $prefix.$alias;
     }
 
     switch ($ts['kind']) {
         case TypeStructureKind::OF_INT:
-            return 'HH\int';
+            return $prefix.'HH\int';
         case TypeStructureKind::OF_STRING:
-            return 'HH\string';
+            return $prefix.'HH\string';
         case TypeStructureKind::OF_BOOL:
-            return 'HH\bool';
+            return $prefix.'HH\bool';
         case TypeStructureKind::OF_VEC:
-            return Str\format('HH\vec<%s>', type_structure_to_type_alias($ts['generic_types'] as nonnull[0]));
+            return $prefix.Str\format('HH\vec<%s>', type_structure_to_type_alias($ts['generic_types'] as nonnull[0]));
         case TypeStructureKind::OF_ENUM:
-        //case TypeStructureKind::OF_UNRESOLVED: // not sure if this is needed
-            return $ts['classname'] as nonnull;
+        case TypeStructureKind::OF_INTERFACE:
```

Comment:
Not sure if we intentionally left this out, but needed this to resolve the `__Type` interface within the `__InputValue`

Reply:
I don't think anything is intentionally omitted here, we just didn't need it so far (also OF_CLASS probably, and OF_UNRESOLVED might be needed, I'm not sure when that applies -- probably when we have something that hasn't been loaded yet so it doesn't know if it's a class/interface/enum, not sure if that can happen).